src/Storage/FileSystem/FileSystem.php

```diff
@@ -105,7 +107,7 @@ public function writeAttempt()
         /**
          * File locking disabled?
          */
-        if (empty($this->file_lock)) {
+        if ($this->file_lock === null) {
```

Comment:
I use "Php Inspections (EA extended)" plugin for PhpStorm, it recommends to replace "empty" with null check. https://github.com/kalessil/phpinspectionsea/blob/master/docs/types-compatibility.md#empty-usage Also I've read some years ago that null check is much faster. I may change it back, up to you.